src/utils/enum.ts

```diff
@@ -0,0 +1,101 @@
+import { extend } from './object';
+
+type ObjectEnum = Record<string, boolean>;
+type Enum = string | ObjectEnum;
+
+/**
+ * Converts from a comma separated object or an array of strings to an object with each string being the property name.
+ */
+function toObject(values: string | string[] | ObjectEnum) {
+    const obj: ObjectEnum = {};
+
+    if (Array.isArray(values)) {
+        values.forEach((value) => {

```

Comment:
I changed for loop to forEach, now I'm wondering whether there were any valid performance reasons why the for loop was used.